src/lib.rs

```diff
@@ -120,12 +120,39 @@ impl<Kind, Type> IdMap<Kind, Type> where Type: Eq + Clone, Kind: Clone {
     }
 }
 
-type SyncExtSender = Mutex<Box<ExtSender<WatchEvent>>>;
-type WatchersMap = HashMap<usize, Arc<SyncExtSender>>;
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum SendDirection {
+    Enter,
+    Exit,
+}
+
+trait RangeChecker {
```

Comment:
About 2/ we can't likely implement it here because these objects are not from our crate.

About 1/ https://github.com/fxbox/openzwave-adapter/issues/13